oracle/src/events/processAffirmationRequests/estimateGas.js

```diff
@@ -4,19 +4,25 @@ const logger = require('../../services/logger').child({
   module: 'processAffirmationRequests:estimateGas'
 })
 
-async function estimateGas({ web3, homeBridge, validatorContract, recipient, value, txHash, address }) {
+async function estimateGas({ web3, homeBridge, validatorContract, recipient, value, nonce, address }) {
   try {
-    return await homeBridge.methods.executeAffirmation(recipient, value, txHash).estimateGas({
+    return await homeBridge.methods.executeAffirmation(recipient, value, nonce).estimateGas({
       from: address
     })
   } catch (e) {
     if (e instanceof HttpListProviderError) {
       throw e
     }
 
-    const messageHash = web3.utils.soliditySha3(recipient, value, txHash)
+    const messageHash = web3.utils.soliditySha3(recipient, value, nonce)
     const senderHash = web3.utils.soliditySha3(address, messageHash)
 
+    // Check if msg is approved by Hashi
+    const isApprovedByHashi = await homeBridge.methods.isApprovedByHashi(messageHash)
```

Comment:
same

Reply:
Solved in [b688329](https://github.com/gnosischain/tokenbridge/pull/5/commits/b688329b9028bdd7d43ee47bddd045ebb0ae2d27)